### PR TITLE
feat(hacbs-1112): add unit testing for pipeline controller & adapter

### PR DESCRIPTION
**Pull request by srivickynesh**
Signed-off-by: srivickynesh <sselvan@redhat.com>

### CODE REVIEW

File: controllers/pipeline/pipeline_adapter_test.go [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		Expect(k8sClient.Status().Update(ctx, testpipelineRunBuild)).Should(Succeed())
+
+		Eventually(func() error {
```
Comment (dirgim) [inline controllers/pipeline/pipeline_adapter_test.go:157]: By changing this Eventually block to also wait for the `testpipelineRunBuild` status to update I managed to make the tests pass reliably. This is what I changed it to:
```
Eventually(func() bool {
			err := k8sClient.Get(ctx, types.NamespacedName{
				Name:      testpipelineRunBuild.Name,
				Namespace: "default",
			}, testpipelineRunBuild)
			return err == nil && len(testpipelineRunBuild.Status.TaskRuns) > 0
		}, time.Second*10).Should(BeTrue())
```

Comment (srivickynesh) [reply controllers/pipeline/pipeline_adapter_test.go]: Done

**Review by dirgim — APPROVED** [review]
LGTM

**Review by hongweiliu17 — APPROVED** [review]
lgtm

**Pull request merged**